addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -364,6 +364,27 @@ export class ProductScreen extends Component {
         );
     }
 
+    getProductCartAmount(product) {
+        const productTmplValIds = product.attribute_line_ids
+            .map((l) => l.product_template_value_ids)
+            .flat();
+
+        const orderLines =
+            productTmplValIds.length > 1
+                ? this.pos.models["product.product"]
+                      .filter((p) => p.raw.product_tmpl_id === product.raw.product_tmpl_id)
+                      .map((p) => p["<-pos.order.line.product_id"])
+                      .flat()
+                : product["<-pos.order.line.product_id"];
+
+        const productAmount = orderLines.reduce(
+            (acc, ol) => (ol.order_id?.uuid === this.currentOrder.uuid ? acc + ol.qty : acc),
+            0
+        );
+
+        return productAmount;
```

Comment:
I believe we can do it easily by checking the `product_template_id` with current order > orderlines > product > `product_tmpl_id`